staging/src/k8s.io/dynamic-resource-allocation/resourceclaim/pod_test.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceclaim
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```

Comment:
In SIG scheduling we prefer not to use assertion libraries. Please use `t.Error`

Reply:
https://github.com/kubernetes/community/blob/master/sig-scheduling/CONTRIBUTING.md#technical-and-style-guidelines

Reply:
As SIG Testing tech lead I don't agree with those guidelines. I accept it for code owned by SIG scheduling, but this here is a general-purpose helper package.

Reply:
Testify keeps us tied to a yaml library we want to drop eventually (https://github.com/stretchr/testify/blob/master/go.mod#L11), it would be vaguely nice to not add more deps to it

Reply:
(But not a blocker, certainly)

Reply:
https://github.com/stretchr/testify/pull/1579 added build tags which avoid that dependency. I'll have a look whether we can leverage that.

Reply:
We can't :-(

go dependency and vendoring don't consider build tags since they have to resolve ahead of when builds are done and be valid for whatever builds *could* be done